configserver/src/main/java/com/yahoo/vespa/config/server/http/v2/response/SessionPrepareAndActivateResponse.java

```diff
@@ -17,25 +18,19 @@
  */
 public class SessionPrepareAndActivateResponse extends SlimeJsonResponse {
 
-    public SessionPrepareAndActivateResponse(PrepareResult result, HttpRequest request, ApplicationId applicationId, Zone zone) {
-        super(result.deployLogger().slime());
+    public SessionPrepareAndActivateResponse(PrepareAndActivateResult result, ApplicationId applicationId) {
+        super(result.prepareResult().deployLogger().slime());
 
         TenantName tenantName = applicationId.tenant();
-        String message = "Session " + result.sessionId() + " for tenant '" + tenantName.value() + "' prepared and activated.";
+        String message = "Session " + result.prepareResult().sessionId() + " for tenant '" + tenantName.value() + "' prepared" +
+                         (result.activationFailure() == null ? " and activated." : ", but activation failed: " + result.activationFailure().getMessage());
         Cursor root = slime.get();
 
-        root.setString("session-id", Long.toString(result.sessionId()));
         root.setString("message", message);
+        root.setString("sessionId", Long.toString(result.prepareResult().sessionId()));
+        root.setBool("activated", result.activationFailure() == null);
 
-        // TODO: remove unused fields, but add whether activation was successful.
-        root.setString("tenant", tenantName.value());
-        root.setString("url", "http://" + request.getHost() + ":" + request.getPort() +
-                              "/application/v2/tenant/" + tenantName +
-                              "/application/" + applicationId.application().value() +
-                              "/environment/" + zone.environment().value() +
-                              "/region/" + zone.region().value() +
-                              "/instance/" + applicationId.instance().value());
```

Comment:
I don't think you can change field names and remove something here, this is a documented API:
https://docs.vespa.ai/en/reference/deploy-rest-api-v2.html#prepareandactivate